keadm/cmd/keadm/app/cmd/util/set.go

```diff
@@ -0,0 +1,501 @@
+	structVal := reflect.ValueOf(structPtr).Elem()
+
+	path := strings.Split(fieldPath, ".")
+	var fieldVal reflect.Value
+	fieldVal = structVal
+	for _, p := range path {
+		fieldVal = getFieldValue(fieldVal, p)
+		if !fieldVal.IsValid() {
+			return fmt.Errorf("%s: No such field: %s in Config", fieldPath, p)
+		}
+	}
+
+	val := reflect.ValueOf(value)
+
+	if fieldVal.Type() != val.Type() {
+		return fmt.Errorf("%s: Provided value type %s does not match field type %s", fieldPath, val.Type(), fieldVal.Type())
+	}
+	fieldVal.Set(val)
+
+	return nil
+}
+
+// GetFieldValue gets the fieldname of the struct.
+func getFieldValue(v reflect.Value, fieldName string) reflect.Value {
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+
+	f := v.FieldByName(fieldName)
+	if !f.IsValid() {
+		return reflect.Value{}
+	}
+	return f
+}
+
+// ParseAndSetArrayValue combines the function of SetArrayValue and ParseFieldPath1.
+func parseAndSetArrayValue(structPtr interface{}, fieldPath string, newValue interface{}) error {
+	path, index, _ := parseFieldPath1(fieldPath)
+	err := setArrayValue(structPtr, path, index, newValue)
+	// fmt.Println(newValue)
+	return err
+}
+
+// SetArrayValue modifies the new value of the name in the config represented by struct.
+// The type of new value may be int, float, string.
+// The name is represented by name1.name2.(...).nameM[N].
+func setArrayValue(structPtr interface{}, fieldPath string, index int, newValue interface{}) error {
+	v := reflect.ValueOf(structPtr).Elem()
+
+	pathParts := strings.Split(fieldPath, ".")
+	var fieldVal reflect.Value
+	fieldVal = v
+	for _, part := range pathParts {
+		fieldVal = getFieldValue(fieldVal, part)
+		// fieldVal = fieldVal.FieldByName(part)
+		if !fieldVal.IsValid() {
+			return fmt.Errorf("%s: No such field: %s in Config", fieldPath, part)
+		}
+	}
+	if fieldVal.Kind() != reflect.Array && fieldVal.Kind() != reflect.Slice && fieldVal.Kind() != reflect.Map {
+		return fmt.Errorf("%s is not an array,slice and map", pathParts[len(pathParts)-1])
+	}
+	// If it is an array or slice, you need to check whether the index is within the range
+	if index < 0 || index >= fieldVal.Len() {
+		return fmt.Errorf("index out of range for %s", pathParts[len(pathParts)-1])
+	}
+
+	//Extend array or slice length
+	if index >= fieldVal.Len() {
+		newSlice := reflect.MakeSlice(fieldVal.Type(), index+1, index+1)
+		reflect.Copy(newSlice, fieldVal)
+		fieldVal.Set(newSlice)
+	}
+	//Set new value
+	elem := reflect.ValueOf(newValue)
+	if elem.Type() != fieldVal.Type().Elem() {
+		return fmt.Errorf("type mismatch for field %s", pathParts[len(pathParts)-1])
+	}
+	fieldVal.Index(index).Set(elem)
+	return nil
+}
+
+// ParseFieldPath1 parses the names in form of "name1.name2.(...).nameM[N]".
+// path : name1.name2.(...).nameM, index : N
+func parseFieldPath1(fieldPath string) (string, int, error) {
+	parts := strings.Split(fieldPath, "[")
+	if len(parts) != 2 {
+		return "", -1, fmt.Errorf("invalid field path")
```

Comment:
```suggestion
		return "", -1, errors.New("invalid field path")
```
You can use the function errors.New() when without formatting. Other similar places also need to be modified

Reply:
I have changed these.